File: project/ranyoung-onboarding/src/main/java/com/example/ranyoungonboarding/api/SurveyResponse.java
package com.example.ranyoungonboarding.api;

import com.example.ranyoungonboarding.domain.Question;
import com.example.ranyoungonboarding.domain.Survey;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Getter
@Builder
public class SurveyResponse {
    private Long id;
    private String name;
    private String description;
    private List<QuestionResponse> questions;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public static SurveyResponse fromEntity(Survey survey) {
        return SurveyResponse.builder()
                .id(survey.getId())
                .name(survey.getName())
                .description(survey.getDescription())
                .questions(survey.getQuestions().stream()
                        .map(QuestionResponse::fromEntity)
                        .collect(Collectors.toList()))
                .createdAt(survey.getCreatedAt())
                .updatedAt(survey.getUpdatedAt())
                .build();
    }
}

@Getter
@Builder
class QuestionResponse {
    private Long id;
    private String name;
    private String description;
    private String type;
    private Boolean required;
    private List<String> options;
    private Integer position;

    public static QuestionResponse fromEntity(Question question) {
        return QuestionResponse.builder()
                .id(question.getId())
                .name(question.getName())
                .description(question.getDescription())
                .type(question.getType().name())
                .required(question.getRequired())
                .options(question.getOptions())
                .position(question.getPosition())
                .build();
    }

}

File: project/hyunwoo-onboarding/presurvey-api/src/main/kotlin/com/innercircle/presurveyapi/api/survey/dto/SurveyResponse.kt
package com.innercircle.presurveyapi.api.survey.dto

import com.innercircle.presurveyapi.domain.survey.model.Question
import com.innercircle.presurveyapi.domain.survey.model.QuestionOption
import com.innercircle.presurveyapi.domain.survey.model.QuestionType
import com.innercircle.presurveyapi.domain.survey.model.Survey
import java.util.*

data class SurveyResponse(
    val id: UUID?,
    val title: String,
    val description: String,
    val questions: List<QuestionResponse>
) {
    companion object {
        fun from(survey: Survey): SurveyResponse {
            return SurveyResponse(
                id = survey.id,
                title = survey.title,
                description = survey.description,
                questions = survey.questions.map { QuestionResponse.from(it) }
            )
        }
    }
}

data class QuestionResponse(
    val id: UUID?,
    val title: String,
    val description: String,
    val type: QuestionType,
    val required: Boolean,
    val options: List<OptionResponse> = emptyList()
) {
    companion object {
        fun from(question: Question): QuestionResponse {
            return QuestionResponse(
                id = null,
                title = question.title,
                description = question.description,
                type = question.type,
                required = question.required,
                options = question.options.map { OptionResponse.from(it) }
            )
        }
    }
}

data class OptionResponse(
    val id: UUID,
    val text: String
) {
    companion object {
        fun from(option: QuestionOption): OptionResponse {
            return OptionResponse(
                id = option.id,
                text = option.text
            )
        }
    }
}